common/tools/dev-tool/src/commands/samples/checkNodeVers.ts

```diff
@@ -0,0 +1,323 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import fs from "fs-extra";
+import path from "path";
+import pr from "child_process";
+import os from "os";
+
+import { createPrinter } from "../../util/printer";
+import { leafCommand, makeCommandInfo } from "../../framework/command";
+import { S_IRWXO } from "constants";
+import { resolveProject } from "../../util/resolveProject";
+
+const log = createPrinter("check-node-vers-samples");
+
+function spawnCMDIgnoreError(
+  cmd: string,
+  args: string[],
+  errorMessage?: string,
+  callback?: () => void
+) {
+  const spawnedProcess = pr.spawn(cmd, args);
+  spawnedProcess.on("error", function() {
+    log.info(errorMessage);
+    callback?.();
+  });
+  spawnedProcess.on("exit", function() {
+    callback?.();
+  });
+}
+
+function deleteDockerContainerCallback(deleteContainerNames?: string[], callback?: () => void) {
+  if (deleteContainerNames) {
+    log.info(`cleanup: deleting ${deleteContainerNames.join(", ")} docker containers`);
+    spawnCMDIgnoreError(
+      "docker",
+      ["rm", ...deleteContainerNames, "-f"],
+      `Attempted to delete ${deleteContainerNames.join(
+        ", "
+      )} docker containers but encountered an error doing so`,
+      callback
+    );
+  } else {
+    callback?.();
+  }
+}
+
+function deleteDockerImageCallback(dockerImageNames?: string[], callback?: () => void) {
+  if (dockerImageNames) {
+    log.info(`cleanup: deleting ${dockerImageNames.join(", ")} docker images`);
+    spawnCMDIgnoreError(
+      "docker",
+      ["rmi", ...dockerImageNames, "-f"],
```

Comment:
IIUC this will delete the alpine node image even if I had it installed before running this command.

Reply:
The user will need to pass `--keep-docker-image` to not delete the images after containers finish running.